filter.go

```diff
@@ -0,0 +1,48 @@
+package files
+
+import (
+	"path/filepath"
+
+	ignore "github.com/crackcomm/go-gitignore"
+)
+
+// Filter - Filter filter.
+type Filter struct {
+	// IncludeHidden - Include hidden files
+	IncludeHidden bool
+	// Rules - File filter rules
+	Rules *ignore.GitIgnore
+}
+
+// NewFilter creates a new file filter form .gitignore and list of rules.
+func NewFilter(ignoreFile string, rules []string, includeHidden bool) (filter *Filter, err error) {
+	filter = &Filter{IncludeHidden: includeHidden}
+	if ignoreFile == "" {
+		filter.Rules, err = ignore.CompileIgnoreLines(rules...)
+	} else {
+		filter.Rules, err = ignore.CompileIgnoreFileAndLines(ignoreFile, rules...)
+	}
+	return
```

Comment:
Nit: I'm not sure we need this return statement here?

Reply:
When using named return values in the type signature, you can omit the variables in the return, but you still need the return statement. I refactored the function to more readable, hopefully.